jsparagus/parse_table.py

```diff
@@ -1477,6 +1479,73 @@ def transform() -> typing.Iterator[None]:
 
         consume(transform(), progress)
 
+    def fold_replay_unwind(self, verbose: bool, progress: bool) -> None:
+        """Convert Replay action falling into Unwind action to an Unwind action which
+        replay less terms."""
+        if verbose or progress:
+            print("Fold Replay followed by Unwind actions.")
+
+        maybe_unreachable_set: OrderedSet[StateId] = OrderedSet()
+
+        def try_transform(s: StateAndTransitions) -> bool:
+            if len(s.epsilon) != 1:
+                return False
+            replay_term, replay_dest_idx = next(iter(s.epsilon))
+            if not isinstance(replay_term, Replay):
+                return False
+            replay_dest = self.states[replay_dest_idx]
+            if len(replay_dest.epsilon) != 1:
+                return False
+            unwind_term, unwind_dest_idx = next(iter(replay_dest.epsilon))
```

Comment:
this doesn't check the class of `unwind_term`.
does it mean that it also expects something else than `Unwind` class?
(the following conditions also implies this may not be `Unwind`, like `update_stack` which is always `True` for `Unwind`)

can you add some comment or check/assert about the following action type?

Reply:
I will do that, note however that we cannot simply check for `Unwind` using `isinstance` as if there is a function call associated, the `Unwind` operation would be the last action of a `Seq` action which will contain one or multiple `FunCall`.